src/Microsoft.Azure.WebJobs.Host/Tables/TableAttributeBindingProvider.cs

```diff
@@ -49,6 +52,143 @@ public TableAttributeBindingProvider(INameResolver nameResolver, IStorageAccount
         }
 
+        // [Queue] has some pre-existing behavior where the storage account can be specified outside of the [Queue] attribute. 
+        // The storage account is pulled from the ParameterInfo (which could pull in a [Storage] attribute on the container class)
+        // Resolve everything back down to a single attribute so we can use the binding helpers. 
+        // This pattern should be rare since other extensions can just keep everything directly on the primary attribute. 
+        private async Task<TableAttribute> CollectAttributeInfo(TableAttribute attrResolved, ParameterInfo parameter, INameResolver nameResolver)
+        {
+            // Look for [Storage] attribute and squirrel over 
+            IStorageAccount account = await _accountProvider.GetStorageAccountAsync(parameter, CancellationToken.None, nameResolver);
+            StorageClientFactoryContext clientFactoryContext = new StorageClientFactoryContext
+            {
+                Parameter = parameter
+            };
+            IStorageTableClient client = account.CreateTableClient(clientFactoryContext);
+
+            return new ResolvedTableAttribute(attrResolved, client);
+        }
+        
+        public static IBindingProvider Build(INameResolver nameResolver, IConverterManager converterManager, IStorageAccountProvider accountProvider, IExtensionRegistry extensions)
+        {
+            var original = new TableAttributeBindingProvider(nameResolver, accountProvider, extensions);
+
+            converterManager.AddConverter<JObject, ITableEntity, TableAttribute>(original.JObjectToTableEntityConverterFunc);
+
+            var bindingFactory = new BindingFactory(nameResolver, converterManager);
+            var bindAsyncCollector = bindingFactory.BindToAsyncCollector<TableAttribute, ITableEntity>(original.BuildFromTableAttribute, null, original.CollectAttributeInfo);
+
+            var bindToJobject = bindingFactory.BindToExactAsyncType<TableAttribute, JObject>(original.BuildJObject, null, original.CollectAttributeInfo);
+            var bindToJArray = bindingFactory.BindToExactAsyncType<TableAttribute, JArray>(original.BuildJArray, null, original.CollectAttributeInfo);
+
+            // Filter to just support JObject, and use legacy bindings for everything else. 
+            // Once we have ITableEntity converters for pocos, we can remove the filter. 
+            // https://github.com/Azure/azure-webjobs-sdk/issues/887
+            bindAsyncCollector = bindingFactory.AddFilter<TableAttribute>(
+                (attr, type) => (type == typeof(IAsyncCollector<JObject>) || type == typeof(ICollector<JObject>)), 
+                bindAsyncCollector); 
+
+            var bindingProvider = new GenericCompositeBindingProvider<TableAttribute>(
+                new IBindingProvider[] { bindToJArray, bindToJobject, bindAsyncCollector, original });
+
```

Comment:
Yes.   The new & old interop nicely since they're all BindingProvders. I filed [https://github.com/Azure/azure-webjobs-sdk/issues/887](https://github.com/Azure/azure-webjobs-sdk/issues/887) to track moving the rest of the table bindings over to the new style (like what we did for Queue) 

---
In reply to: [85971559](https://github.com/Azure/azure-webjobs-sdk/pull/891#discussion_r85971559) [](ancestors = 85971559)